lib/transforms/import-declarations.js

```diff
@@ -28,6 +28,8 @@ function transformer(file, api) {
           var inTestsPrefix = appName + '/app/tests/';
           if (importPath.startsWith(inTestsPrefix)) {
             importPath = appName + '/tests/' + importPath.substring(inTestsPrefix.length);
+          } else {
+            importPath = importPath.replace('/app/app/', '/app/');
```

Comment:
Yea, there's too much magic - I hate how i hardcoded config into the makeAbsoluteHelper. I'm working on a slight refactoring that should make things saner (basically the `TODO: extract these resolutions into classic-file-info and test-file-info`) and will try your updated testcases on that

But dont wait on me if you want to move this forward